### PR TITLE
Make reagent containers like fuel tanks and biomatter tanks refilable, creating overlay of biomatter tube for biomatter solidifier

### PR DESCRIPTION
**Pull request by AlexMorgan3817**
## About The Pull Request
<details><summary>About The Pull Request</summary>

    Now u can refill fuel, water and etc, by beaker of botle by MouseDrop on it
</details>

## Why It's Good For The Game
<details><summary>Why It's Good For The Game</summary>

    Traitors can infect by virus, corrupt by something watter in watter tanks
    and you will needn't to waste this tanks cuz of "It isn't refilable, why i need to store it?"
</details>

## Changelog
<details><summary>Changelog</summary>

:cl:
tweak:  Now you can refill fuel, water and etc, by beaker of botle by MouseDrop on it
imageadd: overlay of biomatter pipe for biomatter solidifier
/:cl:

</details>

### CODE REVIEW

File: code/modules/reagents/holder.dm [hunk 1]
```diff
@@ -369,7 +369,7 @@
 //not directly injected into the contents. It first calls touch, then the appropriate trans_to_*() or splash_mob().
 //If for some reason touch effects are bypassed (e.g. injecting stuff directly into a reagent container or person),
 //call the appropriate trans_to_*() proc.
-/datum/reagents/proc/trans_to(datum/target, amount = 1, multiplier = 1, copy = 0)
+/datum/reagents/proc/trans_to(datum/target, amount = 1, multiplier = 1, copy = 0, ignore_isinjectable = 0)
```
Comment (LethalGhost) [inline code/modules/reagents/holder.dm]: That way seems kinda tricky

Comment (LethalGhost) [reply code/modules/reagents/holder.dm]: Why not just update `is_injectable(...)` proc for tanks?

Comment (LethalGhost) [reply code/modules/reagents/holder.dm]: Or redefine tanks `attackby`

Comment (AlexMorgan3817) [reply code/modules/reagents/holder.dm]: > Or redefine tanks `attackby`

No redefine cuz of attackby must fill beaker, and mousedrop will not fill it, i think it's better than alert with "You want to fill beaker or empty it in this tank?"

**Comment by ITCups** [comment]
have you tried to fiddle with atom's reagent flags ?
`// Atom reagent_flags
#define INJECTABLE		(1<<0)	// Makes it possible to add reagents through droppers and syringes.
#define DRAWABLE		(1<<1)	// Makes it possible to remove reagents through syringes.

#define REFILLABLE		(1<<2)	// Makes it possible to add reagents through any reagent container.
#define DRAINABLE		(1<<3)	// Makes it possible to remove reagents through any reagent container.

#define TRANSPARENT		(1<<4)	// For containers with fully visible reagents.
#define AMOUNT_VISIBLE	(1<<5)	// For non-transparent containers that still have the general amount of reagents in them visible.
`

**Comment by AlexMorgan3817** [comment]
@ITCups, why i need it, i just did injectable flag bypass, cuz in attackby we fill beaker, but in MouseDrop we do nothing with beaker and i insert in MouseDrop emptying of beaker in dispenser

**Comment by AlexMorgan3817** [comment]
Ready for review

**Comment by AlexMorgan3817** [comment]
ready4review

**Comment by LethalGhost** [comment]
> If any of you want to change something there, just make a new PR, we need this to work for now.

Nothing lasts longer than the temporary

**Pull request merged**